playwright/tests/page-objects/inscription.ts

```diff
@@ -235,46 +151,82 @@ class Select {
 
 export class Input {
   readonly locator: Locator;
+  readonly outputLocator: Locator;
+  readonly inputLocator: Locator;
 
   constructor(
     readonly page: Page,
     readonly parentLocator: Locator,
-    options?: { label?: string; nth?: number; type?: string }
+    options?: { label?: string; nth?: number; type?: 'number' | 'id' }
   ) {
     const role = options?.type === 'number' ? 'spinbutton' : 'textbox';
-    if (options?.label) {
-      this.locator = parentLocator.getByRole(role, { name: options.label }).first();
+    const badgeLocator = parentLocator.locator('.badge-field');
+    if (role === 'spinbutton' || options?.type === 'id') {
```

Comment:
sounds strange to me that you first create this role based on the options.type and then use the type here again for the check...

Reply:
I think I wouldn't save the role to a variable, but directly use the string in the if else blocks...

Reply:
The role variable is needed to determine if it is a textbox or a spinbutton, but i've simplifyed the condition